content/golang/docker/publish-images-multi-arch.md

```diff
@@ -15,45 +15,402 @@ tags:
+
+{{< alert "info" >}}
+If you are not familiar with Dockerfiles, you can read more about them <a href="https://docs.docker.com/engine/reference/builder/">here</a>.
+{{< / alert >}}
+
+
+{{< alert "info" >}}
+We could omit the second stage, but a Docker image should not be larger than necessary, and we do not need the Go build environment inside the final image when distributing it.
+{{< / alert >}}
+
+Create a `Dockerfile` with the following content:
+
+{{< highlight text "linenos=table,hl_lines=99" >}}
+# Use the official Go image to build the application with.
+FROM golang:1.11 as builder
+
+# Create a working directory.
+WORKDIR /hello
+
+# Add the files necessary for building the application.
+ADD main.go /hello
+ADD go.mod /hello
+
+# Then set the Go build environment variables.
+
+# We need this because if it is omitted,
+# the final binary will be linked to libc,
+# and will have dependency issues on Alpine.
+ENV CGO_ENABLED 0
+
+# This is the target Operating system,
+# we use "linux" here because it is the most commonly used,
+# but we could also use "windows" (E.g. targetting Windows 10 IoT Core).
+ENV GOOS linux
+
+# Our target architecture, ARM64 in this case.
+ENV GOARCH arm64
+
+# Build the application,
+# this will create a "hello-world" binary.
+RUN go build
+
+# Create the second stage, the final image.
+# Make sure you use the correct target architecture
+# of the base image, ARM64 in this case.
+FROM arm64v8/alpine
+
+# We use the same working directory for convenience.
+WORKDIR /hello
+
+# Copy the already built binary file from the first stage.
+COPY --from=builder /hello/hello-world .
+
+# Our application uses the 8080 port
+# to communicate, so we need to expose it.
+EXPOSE 8080
+
+# Finally we set the entry point of the image,
+# so that the hello-world server runs whenever the container is started.
+ENTRYPOINT [ "/hello/hello-world" ]
+{{< / highlight >}}
+
+Each step has comments in the `Dockerfile`, make sure you read them, and understand what is happening.
+
+With our `Dockerfile` in place, we can finally build the image with Docker.
+
+{{< alert "info" >}}
+We also tag the image "hello-world" for convenience.
+{{< / alert >}}
+
+
+```
+docker build -t hello-world .
+```
+
+After the code has run, you should see the `hello-world` image when running `docker image ls`.
+
+On linux, simply use:
+```
+docker image ls | grep hello-world
+```
+
+You should a similar output:
+```
+hello-world latest 8fd45a7d5056 2 minutes ago 11.5MB
+```
+
+Now you can use the image on your ARM64 devices, and/or publish it to [Docker Hub](https://docs.docker.com/docker-hub/).
+
+{{< alert "info" >}}
+Note that because the architecture of the image is ARM64, you cannot natively run it on AMD64 machines.
```

Comment:
note thate you can run arm64 images on Docker for Mac and Docker for Windows. The Docker distribution includes qemu for this and I know arm will want to highlight this :)

Reply:
Ah thanks, fair enough, I'll update it with that info and any additional instructions if needed. I won't be able to test it on mac though.

What about the general feel/direction of the guide?